tests/Traits/SubEntityReferenceTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content\Traits;
+
+use Drupal\Tests\oe_content\Behat\Content\Traits\GatherSubEntityContextTrait;
```

Comment:
Yeah, we are mixing concerns here, thing is: normal entities have labels, sub-entities do not, so the problem here is in the method signature:

```php
   protected function getSubEntityReferenceField(string $field_name, string $labels): array {
```

We shouldn't require labels there but, instead, the actual sub-entity object, then however we get that object, is not a concern of the `SubEntityReferenceTrait` trait, which means we will use the other trait for that, outside, in the called code, and pass the returned entity objects in this method.

The new signature should be something like:

```php
   protected function getSubEntityReferenceField(string $field_name, SubEntityInterface $entities) array {
```